packages/modular-scripts/react-scripts/config/getHttpsConfig.js

```diff
@@ -1,10 +1,12 @@
 'use strict';
 
+require('ts-node').register();
```

Comment:
I think @steveukx was referring to running the project with `node -r ts-node/register`, not using it programatically. I run it with `ts-node` directly, which does the smae preloading and should allow you to import ts files in your js files:

`alias modular-dev="~/modular/node_modules/.bin/ts-node ~/modular/packages/modular-scripts/src/cli.ts"`

There is an important caveat, though: if you try to require ts files in a js file that is loaded by Webpack (for example, anything required by webpack.config.js), this won't work, because it will use a different instance of node spawned directly by Webpack. In this case, `getHttpsConfig` is required by `start`, so it should work.

(for Webpack files, you can use `TS_NODE_PROJECT`, as @steveukx just discovered).

Reply:
Ah okay - I thought all our scripts already ran with ts-node, for example tests: `ts-node packages/modular-scripts/src/cli.ts test --watchAll false --runInBand --env node` - and that isn't happy with importing a ts file in js, while when running it programmatically it didn't complain (tho I'm still facing other problems)

We can chat more about it, and see what to do for Webpack as well

Reply:
Ignore all of the above I went down the wrong rabbit hole.
I'm still unclear on what the 'right' way of importing a typescript function in a javascript file is

Reply:
@AlbertoBrusa just for fun, you can also try to use tsx (which in turn uses esbuild transform) instead of ts-node. I don't know if it solves your problem, but worth checking: https://github.com/esbuild-kit/tsx

Reply:
@AlbertoBrusa you can also try ts-node in ESM mode -> https://github.com/TypeStrong/ts-node#esm (if you haven't already)

Reply:
Finally, people from the internet say that ignoring warnings might work -> `node --no-warnings --loader ts-node/esm file.ts`

Reply:
Thanks, I'll give all these approaches a try tomorrow